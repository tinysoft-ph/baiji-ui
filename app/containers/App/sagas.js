import { takeLatest } from 'redux-saga/effects';

import {
  ME_FROM_TOKEN,
} from './constants';

function* getMeFromToken() {

}

function* meData() {
  yield takeLatest(ME_FROM_TOKEN, getMeFromToken);
}

export default [
  meData,
];
