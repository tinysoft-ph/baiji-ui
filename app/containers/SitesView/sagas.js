import { takeLatest, call, put } from 'redux-saga/effects';

import request from 'utils/request';

import {
  LOAD_SITE,
} from './constants';

import {
  loadSiteViewSuccess,
} from './actions';

export function* loadSiteView(id) {
  const url = `/site?id=${id}`;

  try {
    const view = yield call(request, url);
    yield put(loadSiteViewSuccess(view));
  } catch (err) {
    yield put(err.message);
  }
}

// Individual exports for testing
export function* getSiteView() {
  yield takeLatest(LOAD_SITE, loadSiteView);
}

// All sagas to be loaded
export default [
  getSiteView,
];
