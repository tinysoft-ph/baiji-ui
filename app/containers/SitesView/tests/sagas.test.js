/**
 * Test  sagas
 */

/* eslint-disable redux-saga/yield-effects */
import { take, call, put, select } from 'redux-saga/effects';
import { loadSiteView } from '../sagas';

import {
  loadSiteViewSuccess,
} from '../actions';

describe('loadSiteView Saga', () => {
  let loadSiteViewGenerator;

  beforeEach(() => {
    loadSiteViewGenerator = loadSiteView({
      id: '1234-abcd',
    });
    const testCall = loadSiteViewGenerator.next().value;

    expect(testCall).toMatchSnapshot();
  });

  it('Expect to have unit tests specified', () => {
    const fixture = {
      siteName: 'test-site',
    };

    const callDescriptor = loadSiteViewGenerator.next(fixture).value;
    expect(callDescriptor).toEqual(put(loadSiteViewSuccess(fixture)));
  });
});
