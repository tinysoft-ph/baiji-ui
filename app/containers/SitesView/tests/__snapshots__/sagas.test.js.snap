exports[`loadSiteView Saga Expect to have unit tests specified 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "/site?id=[object Object]",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;
